Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -724,9 +729,7 @@ namespace Babylon
         ProgramData* program = new ProgramData{};
         try
         {
-            std::pair<bgfx::ProgramHandle, std::unordered_map<std::string, uint32_t>> programInfo = CreateProgramInternal(vertexSource, fragmentSource, *program);
-            program->Handle = programInfo.first;
-            program->VertexAttributeLocations = std::move(programInfo.second);
+            *program = std::move(*CreateProgramInternal(vertexSource, fragmentSource).release());
```

Comment:
If you use the same pattern as the async version, you shouldn't have to call `release`.

Reply:
So can I just remove `.release()` from this code? The pattern already looks basically the same except for that.

Reply:
Not exactly. Because the try/catch below can throw which will bypass the js object owning the pointer. We need to move the creation of the js object right after `new ProgramData`, just like the async version.

Reply:
Sorry, I missed the delete call in the catch. What you have should work, but the `delete` would be unnecessary if we use the pattern similar to the async code.